django/cantusdb_project/main_app/views/chant.py

```diff
@@ -776,56 +798,76 @@ def get_queryset(self) -> QuerySet:
                 # as a substring
                 feasts = Feast.objects.filter(name__icontains=feast)
                 q_obj_filter &= Q(feast__in=feasts)
-            if self.request.GET.get('order'):
-                if self.request.GET.get('order') == 'siglum':
-                    order = 'siglum'
-                elif self.request.GET.get('order') == 'incipit':
+            order: str = 'siglum'
+            order_get_param: Optional[str] = self.request.GET.get('order')
+            sort_get_param: Optional[str] = self.request.GET.get('sort')
+            if order_get_param is not None:
+                if order_get_param == 'siglum':
+                    pass # this is the default
+                elif order_get_param == 'incipit':
                     order = 'incipit'
-                elif self.request.GET.get('order') == 'office':
+                elif order_get_param == 'office':
                     order = 'office'
-                elif self.request.GET.get('order') == 'genre':
+                elif order_get_param == 'genre':
```

Comment:
Looking a bit more closely at this... this whole block can be removed and mostly replaced with a couple simple checks.

1. Check if the incoming value is in a list of acceptable values
1a. Check the special cases where the incoming value isn't the value we want
1b. Otherwise, set the order value to the incoming value
2. If it isn't in a list of acceptable values, all other values get set to "siglum" (default), including no value.

```py
order_get_param: Optional[str] = self.request.GET.get("order")
if order_get_param in ('incipit', 'office', 'genre', 'cantus_id', 'mode', 'has_fulltext', 'has_melody', 'has_image'):
    if order_get_param == "has_fulltext":
         order = "manuscript_full_text"
    elif order_get_param == "has_melody":
         order = "volpiano"
    elif order_get_param == "has_image":
         order = "image_link"
    else:
         order = order_get_param
else:
    order = 'siglum'
```

Reply:
I like this suggestion.